lib/model/model.go

```diff
@@ -1820,10 +1825,7 @@ func (m *model) AddConnection(conn connections.Connection, hello protocol.HelloR
 	l.Infof(`Device %s client is "%s %s" named "%s" at %s`, deviceID, hello.ClientName, hello.ClientVersion, hello.DeviceName, conn)
 
 	conn.Start()
-	m.pmut.Unlock()
 
-	// Acquires fmut, so has to be done outside of pmut.
-	cm := m.generateClusterConfig(deviceID)
 	conn.ClusterConfig(cm)
```

Comment:
Only thing that makes me a little but uneasy about this is that ClusterConfig() is a potentially blocking call... Although, at this point, the writer loop should be up since Start() and be reading the outbox channel that ClusterConfig() needs to send on, so hopefully it's fine.

Reply:
Good point, in the worst case it blocks `pmut` for as long as the timeout of the underlying connection. Alternative idea: First check and drop old connection holding `pmut`, then release it and already start and send cluster config on the new connection before acquiring `pmut` again and adding the connection to the model. This relies on `AddConnection` not being called concurrently, but that's already the case today. I'll add a comment to that effect, if we want to enfore/ensure it, we'd need to add another lock.